File: countcharactor.c
#include <stdio.h>
#include <string.h>

int main() {
    char num[1000]; // Declare a character array to store the input string
    int count[10] = {0}; // Initialize an array to store the frequency of each digit from 0 to 9

    // Read the input string
    scanf("%s", num);

    // Calculate the length of the input string
    int len = strlen(num);

    // Iterate through each character in the input string
    for (int i = 0; i < len; i++) {
        // Check if the current character is a digit
        if (num[i] >= '0' && num[i] <= '9') {
            // If it's a digit, convert it to an integer and update the count array
            int digit = num[i] - '0';
            count[digit]++;
        }
    }

    // Print the frequency of each digit from 0 to 9
    for (int i = 0; i < 10; i++) {
        printf("%d ", count[i]);
    }
    printf("\n");
    return 0;
}

File: countcharactor.java
import java.util.Scanner;

public class countcharactor {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String num = scanner.nextLine();
        int[] count = new int[10];

        for (int i = 0; i < num.length(); i++) {
            char currentChar = num.charAt(i);
            if (Character.isDigit(currentChar)) {
                int digit = Character.getNumericValue(currentChar);
                count[digit]++;
            }
        }

        for (int i = 0; i < 10; i++) {
            System.out.print(count[i] + " ");
        }
        System.out.print("\n");
    }
}

